### PR TITLE
Command to pull data for Markdown table

### PR DESCRIPTION
**Pull request by wsmelton**
Basic command that pulls the data for the following table format:

| Order | Issue Num | By | Title | Standby |
| --------- | -------- | ------- | -------- | ----- |
| 01 | #2 | @potatoqualitee | Default Parameter Values | No |
| 02 | #4 | @dfinke | The dime tour: PowerShell + Excel = Better Together | No |
| 03 | #5 | @mattcushing | Import Excel files to SQL Server in less than 20 lines of code, seriously!! | No |
| 04 | #6 | @awickham10 | Setting Trace Flags as Startup Parameters with dbatools | No |
| 05 | #7 | @jpomfret | Applying SQL Server Data Compression with dbatools | No |
| 06 | #8 | @alevyinroc | Better, Safer SQL Queries from PowerShell | No |
| 07 | #10 | @devblackops | ChatOps with Microsoft Teams and PoshBot | Yes |
| 08 | #11 | @Windos | Easy Desktop Notifications with BurntToast | Yes |
| 09 | #12 | @DanielSSilva | Raspberry Pi with powershell and IoT module | Yes |
| 10 | #13 | @wsmelton | Getting Started with Visual Studio Code | Yes |

Right off it does not order the output like above so something to work on further.

### CODE REVIEW

File: New-PSPowerHour.ps1 [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        Order = "{0:00}" -f $sequence
+        IssueNumber = "#$($_.Number)"
+        By          = "@$($_.user.login)"
+        Title       = $_.Title
+        StandBy     = if ($sequence -gt 5) {"Yes"} else {"No"}
+    }
+    $sequence++
+} | Select-Object Order, IssueNumber, By, Title, StandBy ConvertTo-Markdown
+
+#ToDo
```
Comment (michaeltlombardi) [inline New-PSPowerHour.ps1]: Suggest removing this line and the next, we can file issues to improve the tooling and keep that info out of the source code. :)

**Comment by wsmelton** [comment]
The output provided in the command:

By              | Order | IssueNumber | StandBy | Title                                                                      
--------------- | ----- | ----------- | ------- | ---------------------------------------------------------------------------
@potatoqualitee | 01    | #2          | No      | Default Parameter Values                                                   
@dfinke         | 02    | #4          | No      | The dime tour: PowerShell + Excel = Better Together                        
@mattcushing    | 03    | #5          | No      | Import Excel files to SQL Server in less than 20 lines of code, seriously!!
@awickham10     | 04    | #6          | No      | Setting Trace Flags as Startup Parameters with dbatools                    
@jpomfret       | 05    | #7          | No      | Applying SQL Server Data Compression with dbatools                         
@alevyinroc     | 06    | #8          | Yes     | Better, Safer SQL Queries from PowerShell                                  
@devblackops    | 07    | #10         | Yes     | ChatOps with Microsoft Teams and PoshBot                                   
@Windos         | 08    | #11         | Yes     | Easy Desktop Notifications with BurntToast                                 
@DanielSSilva   | 09    | #12         | Yes     | Raspberry Pi with powershell and IoT module                                
@wsmelton       | 10    | #13         | Yes     | Getting Started with Visual Studio Code

File: New-PSPowerHour.ps1 [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+$data | ForEach-Object {
+    [PSCustomObject][ordered]@{
+        Order = "{0:00}" -f $sequence
+        IssueNumber = "#$($_.Number)"
+        By          = "@$($_.user.login)"
+        Title       = $_.Title
+        StandBy     = if ($sequence -gt 5) {"Yes"} else {"No"}
+    }
+    $sequence++
+} | Select-Object Order, IssueNumber, By, Title, StandBy ConvertTo-Markdown
```
Comment (michaeltlombardi) [inline New-PSPowerHour.ps1]: Should there be a `|` between `Standby` and `ConvertTo-Markdown`?

Comment (wsmelton) [reply New-PSPowerHour.ps1]: Yeah, caught that when I ran it locally. Already pushed update to fix that one.

File: New-PSPowerHour.ps1 [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    .NOTES
+        This utilizes PowerShellForGitHub module by the PowerShell team
+        For best results and the way written you will need a token from GitHub (no permissions needed to be assigned)
+#>
+[cmdletbinding()]
+param(
+    [string]$GitHubToken
+)
+
+function ConvertTo-Markdown {
```
Comment (michaeltlombardi) [inline New-PSPowerHour.ps1]: Is there any objection to using the [`PSMarkdown`](https://github.com/ishu3101/PSMarkdown) module here instead?

File: New-PSPowerHour.ps1 [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        Order = "{0:00}" -f $sequence
+        IssueNumber = "#$($_.Number)"
+        By          = "@$($_.user.login)"
+        Title       = $_.Title
+        StandBy     = if ($sequence -gt 5) {"Yes"} else {"No"}
+    }
+    $sequence++
+} | Select-Object Order, IssueNumber, By, Title, StandBy | ConvertTo-Markdown
+
+#ToDo
```
Comment (michaeltlombardi) [inline New-PSPowerHour.ps1]: Suggest removing this line and the next, we can file issues to improve the tooling and keep that info out of the source code. :)

File: New-PSPowerHour.ps1 [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+# Order of columns is being messed with in the custom function to convert the table to markdown. Have not reviewed fully to get that straighted out
```
Comment (michaeltlombardi) [inline New-PSPowerHour.ps1]: Where possible, it's best to leave a trailing newline in code files for folks who might run them on not-Windows.

File: New-PSPowerHour.ps1 [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    .NOTES
+        This utilizes PowerShellForGitHub module by the PowerShell team
+        For best results and the way written you will need a token from GitHub (no permissions needed to be assigned)
+#>
+[cmdletbinding()]
+param(
+    [string]$GitHubToken
+)
+
+if (-not (Get-Module PowerShellForGitHub)) {
```
Comment (michaeltlombardi) [inline New-PSPowerHour.ps1]: Should this also use the `ListAvailable` parameter as in the second example? Could this be in a loop to reduce duplicate code?

File: New-PSPowerHour.ps1 [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+}
+if (-not (Get-Module PSMarkdown -ListAvailable)) {
+    Write-Output "Please install PSMarkdown module!"
+}
+
+try {
+    Import-Module PowerShellForGitHub, PSMarkdown
+}
+catch {
+    Write-Warning "Issue loading modules"
```
Comment (michaeltlombardi) [inline New-PSPowerHour.ps1]: Should this throw? None of the rest of the script will work, right?

File: New-PSPowerHour.ps1 (hunk 5 above)
Comment (wsmelton) [reply New-PSPowerHour.ps1]: First I've heard of this...none of the functions we maintain in dbatools has a trailing newline but I know we have non-windows devs and users.

File: New-PSPowerHour.ps1 [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+$data | ForEach-Object {
+    [PSCustomObject][ordered]@{
+        Order       = "{0:00}" -f $sequence
+        IssueNumber = "#$($_.Number)"
+        By          = "@$($_.user.login)"
+        Title       = $_.Title
+        StandBy     = if ($sequence -gt 5) {"Yes"} else {"No"}
+    }
+    $sequence++
+} | Select-Object Order, IssueNumber, By, Title, StandBy | ConvertTo-Markdown
```
Comment (RamblingCookieMonster) [inline New-PSPowerHour.ps1]: Looks like you can just scrub the final Select-Object

File: New-PSPowerHour.ps1 [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    else {
+        Import-Module $m
+    }
+}
+
+$data = Get-GitHubIssueForRepository -repositoryUrl 'https://github.com/pspowerhour/pspowerhour' -gitHubAccessToken $GitHubToken |
+    Where-Object {$_.labels.name -eq 'proposal'} | Select-Object Number, Title, user | sort-object Number | select-object -first 10
+[int]$sequence = 1
+$data | ForEach-Object {
+    [PSCustomObject][ordered]@{
```
Comment (RamblingCookieMonster) [inline New-PSPowerHour.ps1]: You only need `[pscustomobject]` here, this is ordered by default

Comment (wsmelton) [reply New-PSPowerHour.ps1]: It actually isn't, that is why I tried adding it but it does not work. When you pass it to the convert it changes the property order.

File: New-PSPowerHour.ps1 (hunk 8 above)
Comment (wsmelton) [reply New-PSPowerHour.ps1]: I tried this to force the property order when it is passed to the convert but it is lost.

File: New-PSPowerHour.ps1 (hunk 9 above)
Comment (RamblingCookieMonster) [reply New-PSPowerHour.ps1]: Ahhh, just took a look... going to poke around other markdown modules, there are a few issues with PSMarkdown

**Pull request closed without merging**